src/activities/components/ActivitySeriesEdit.vue

```diff
@@ -478,29 +494,81 @@ export default {
     smallScreen () {
       return this.$q.screen.width < 450 || this.$q.screen.height < 450
     },
+    rruleObject () {
+      const { rule, startDate } = this.edit
+      return rrulestr(serializeRule(rule), { dtstart: startDate })
+    },
+    previewActivity () {
+      if (!this.showPreview) return null
+      const {
+        activityType,
+        participantTypes,
+        place,
+        description,
+        startDate,
+        datesPreview = [],
+        duration,
+      } = this.edit
+      const date = datesPreview[0] || this.rruleObject.after(new Date()) || startDate || new Date()
+      const dateEnd = duration ? addSeconds(date, duration) : null
+      return {
+        activityType,
+        participantTypes,
+        place,
+        description,
+        date,
+        dateEnd,
+        hasDuration: Boolean(duration),
+        participants: [],
+      }
+    },
+  },
+  watch: {
+    // TODO: would be nice to include helpful message somewhere, but we access the activities differently now
+    // isPending (val) {
+    //   const hasExceptions = () => {
+    //     const { activities } = this.edit
+    //     return activities.some(({ seriesMeta }) => seriesMeta.isDescriptionChanged || seriesMeta.isMaxParticipantsChanged || !seriesMeta.matchesRule)
+    //   }
+    //   if (!val && !this.hasAnyError && hasExceptions()) {
+    //     Dialog.create({
+    //       title: this.$t('CREATEACTIVITY.EXCEPTIONS_TITLE'),
+    //       message: this.$t('CREATEACTIVITY.EXCEPTIONS_MESSAGE', { upcomingLabel: this.$t('ACTIVITYMANAGE.UPCOMING_ACTIVITIES_IN_SERIES') }),
+    //       ok: this.$t('BUTTON.YES'),
+    //     })
+    //   }
+    // },
   },
   methods: {
     toggleDuration () {
       this.hasDuration = !this.hasDuration
     },
-    maybeSave () {
+    async maybeSave () {
       if (!this.canSave) return
-      this.save()
-    },
-    getCreateData () {
-      return {
-        ...this.edit,
-        activityType: this.activityType.id,
-      }
+      const { participants } = await activitySeriesAPI.checkSave({ ...this.getPatchData(), id: this.value.id })
```

Comment:
this should do something when it errors, ideally show it to the user...

Reply:
it doesn't have a defined error response... any error would be a server error. it always returns 200. if the participants array is empty then it means nobody will be removed if you go ahead, if there are entries in it, it means those would be removed if you go ahead.